### ISSUE TITLE
MAL tracking Completed" status doesn't update "volumes"

### ISSUE THREAD

**Issue by BaronKrause** [inorichi/tachiyomi#1276]
I notice that any manga that gets set to completed either manually or due to the chapter tracking auto setting it to completed, it leaves the volumes blank.

On MAL they have separate volume and chapter tracking, and when you simply mark a volume as completed and it had 0/17 chapters and 0/3 volumes, it will change both the chapter and volumes to 17/17 and 3/3 respectively. When Tachiyomi sets it to completed it will show as 17/17 and 0/3.

No big deal since the important  part is chapter/reading/completed  status along with the rating and I imagine accurately tracking volume status would be a nightmare since not every source even lists it. Honestly no one even uses the volume tracking and it just exists as something  to go from 0 to maxed when you  hit complete, think it would it be possible if once a manga is completed that they tracking just maxes out both like the site?

**Comment by inorichi** [comment]
We don't store volume information because there's no way to track them given the chapter titles.

Implementing this would require a read from the API before a write, besides having to store the volume in memory.

**Comment by Vesnyx** [comment]
I don't think he's referring to the tracking of volumes while reading. 

The current functionality on the MAL website is when a series is marked as complete, the volume is automatically maxed. In the app, it only updates the chapter number and leaves the volume's at 0.

**Comment by inorichi** [comment]
Yes, that's what I understood. Also the reason I said we have to make a read from the API to get the number of volumes.

**Comment by Vesnyx** [comment]
Ah, I misunderstood what you said. Is it possible to parse the details and send the total volumes when the update function recognizes that the manga is completed?

**Comment by inorichi** [comment]
Yes, but before that the whole tracking implementation needs a refactor, but I never have enough motivation to work on it. 

Any change in the tracking tab overrides the values in remote because we send everything we track. If the local data is out of sync (say, for example, you update the score in the webpage and don't pull down the list to update the data, then when we update the last chapter read, the score is reverted to the locally stored).

There should be a method in the interface for each single change (score, last chapter read, status...) and maybe an additional one to mark the manga as completed. Maybe it'd be reasonable to make a read and sync the local data with remote before any write (where we could read the number of volumes).

I'd like to embrace data immutability with the refactor, as the current implementation is a port of the old Java based tachiyomi.

**Comment by scb261** [comment]
I'm not sure if it counts as duplicate of #505 since this issue is mentioned in it
> And even on completed manga when you set them as completed volume count stays at zero

If not, then I guess this issue should have 1.x tag as well
